go/vt/vtgate/planbuilder/operators/aggregation_pushing.go

```diff
@@ -115,37 +115,105 @@ func pushDownAggregationThroughRoute(
 
 // pushDownAggregations splits aggregations between the original aggregator and the one we are pushing down
 func pushDownAggregations(ctx *plancontext.PlanningContext, aggregator *Aggregator, aggrBelowRoute *Aggregator) error {
-	for i, aggregation := range aggregator.Aggregations {
-		if !aggregation.Distinct || exprHasUniqueVindex(ctx, aggregation.Func.GetArg()) {
-			aggrBelowRoute.Aggregations = append(aggrBelowRoute.Aggregations, aggregation)
+	var distinctExpr sqlparser.Expr
+	var differentExpr *sqlparser.AliasedExpr
+
+	distinctAggrPushedDown := true
+	distinctAggrGroupByAdded := false
+
+outer:
+	for i, aggr := range aggregator.Aggregations {
+		if !aggr.Distinct {
+			aggrBelowRoute.Aggregations = append(aggrBelowRoute.Aggregations, aggr)
 			aggregateTheAggregate(aggregator, i)
 			continue
 		}
-		innerExpr := aggregation.Func.GetArg()
-
-		if aggregator.DistinctExpr != nil {
-			if ctx.SemTable.EqualsExpr(aggregator.DistinctExpr, innerExpr) {
-				// we can handle multiple distinct aggregations, as long as they are aggregating on the same expression
-				aggrBelowRoute.Columns[aggregation.ColOffset] = aeWrap(innerExpr)
-				continue
+		funcExpr := aggr.Func.GetArg()
+		switch aggr.OpCode {
+		case opcode.AggregateCountDistinct, opcode.AggregateSumDistinct:
+			if !exprHasUniqueVindex(ctx, funcExpr) {
```

Comment:
we should extract this into a method. deep indentation is bad, m'kay?

![image](https://github.com/vitessio/vitess/assets/402248/109bffc1-3474-4788-a110-de1273517fc8)